src/System.Collections/src/System/Collections/Generic/Queue.cs

```diff
@@ -183,15 +182,31 @@ void ICollection.CopyTo(Array array, int index)
         // Adds item to the tail of the queue.
         public void Enqueue(T item)
         {
-            if (_size == _array.Length)
+            int tail = _tail;
+            int size = _size;
+            T[] array = _array;
+
+            if ((uint)size < (uint)array.Length && (uint)tail < (uint)array.Length)
```

Comment:
doesn't `(uint)tail < (uint)array.Length`   just move  the check to C# code?

Reply:
Correct. The only mini-advantage is that the JITed code size is a few bytes smaller.
```diff
G_M9243_IG02:
       FF471C               inc      dword ptr [rdi+28]
       8B4714               mov      eax, dword ptr [rdi+20]
       8B5718               mov      edx, dword ptr [rdi+24]
       488B4F08             mov      rcx, gword ptr [rdi+8]
       448B4108             mov      r8d, dword ptr [rcx+8]
       443BC2               cmp      r8d, edx
       7620                 jbe      SHORT G_M9243_IG05
-      443BC0               cmp      r8d, eax
-      761B                 jbe      SHORT G_M9243_IG05
+      413BC0               cmp      eax, r8d
+      7323                 jae      SHORT G_M9243_IG07
       4C63C8               movsxd   r9, eax
       4289748910           mov      dword ptr [rcx+4*r9+16], esi
       FFC0                 inc      eax
       443BC0               cmp      r8d, eax
       7502                 jne      SHORT G_M9243_IG03
       33C0                 xor      eax, eax

G_M9243_IG03:
       894714               mov      dword ptr [rdi+20], eax
       FFC2                 inc      edx
       895718               mov      dword ptr [rdi+24], edx

G_M9243_IG04:
       5D                   pop      rbp
       C3                   ret      

G_M9243_IG05:
       E8E0F4FFFF           call     Queue`1:EnqueueWithResize(int):this
       90                   nop      

G_M9243_IG06:
       5D                   pop      rbp
       C3                   ret      

+G_M9243_IG07:
+      E8882CB778           call     CORINFO_HELP_RNGCHKFAIL
+      CC                   int3     
-; Total bytes of code 67, prolog size 5 for method Queue`1:Enqueue(int):this
+; Total bytes of code 73, prolog size 5 for method Queue`1:Enqueue(int):this
; ============================================================
```

I think this can be reverted, thus improving readability of the code.